### DISCUSSION TITLE
How can I match any non numeric including decimals?

### DISCUSSION THREAD

**Discussion by deathwishdave** [taocpp/PEGTL#226] [Q&A] [answered]
How can I match any non numeric including decimals?

`pegtl::not_one` will not allow me to input `pegtl::digit`

**Comment by d-frey** [comment, marked as answer]
You can use [`pegtl::not_range<'0','9'>`](https://github.com/taocpp/PEGTL/blob/master/doc/Rule-Reference.md#not_range-c-d-).

**Comment by deathwishdave** [comment]
thanks
